frontend/src/utils/action.tsx

```diff
@@ -23,16 +25,35 @@ export const CompareNowAction = ({ videoId }: { videoId: string }) => {
 
 export const AddToRateLaterList = ({ videoId }: { videoId: string }) => {
   const video_id = videoId;
+  const { enqueueSnackbar } = useSnackbar();
   return (
     <Tooltip title="Rate later" placement="left">
       <IconButton
         size="medium"
         color="secondary"
-        onClick={() =>
-          UsersService.usersMeVideoRateLaterCreate({
-            video: { video_id } as Video,
-          })
-        }
+        onClick={async () => {
+          let flag = false;
+          const rateLaterList = await UsersService.usersMeVideoRateLaterList();
+          rateLaterList.results?.forEach((rateLaterVideo) => {
+            if (rateLaterVideo.video.video_id === video_id) {
+              flag = true;
+            }
+          });
+          if (!flag) {
+            await UsersService.usersMeVideoRateLaterCreate({
```

Comment:
`UsersService.usersMeVideoRateLaterCreate()` will raise an `ApiError` with status 409, if the video already exists in the rate later list. So calling `UsersService.usersMeVideoRateLaterList()` should not be necessary (and rigorously several pages may be necessary to fetch the whole rate later list, which would be impractical).  

And by catching the API error, a single request to API will be executed.